source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import * as Adaptive from "adaptivecards";
+import { HostContainer } from "../host-container";
+import * as hostConfig from "../../hostConfigs/federated-search.json";
+
+export class FederatedSearchContainer extends HostContainer {
+  constructor(name: string, styleSheet: string) {
+    super(name, styleSheet);
+
+    this.actionsRegistry.unregister("Action.Submit");
+    this.elementsRegistry.unregister("Input.Text");
+    this.elementsRegistry.unregister("Input.Date");
+    this.elementsRegistry.unregister("Input.Time");
+    this.elementsRegistry.unregister("Input.Toggle");
+    this.elementsRegistry.unregister("Input.ChoiceSet");
+    this.elementsRegistry.unregister("Input.Number");
+    this.actionsRegistry.register("Action.Http", Adaptive.HttpAction);
+    this.cardHost.classList.add("fedsearch-card");
+  }
+
+  public renderTo(hostElement: HTMLElement) {
+    hostElement.classList.add("outlook-frame");
+    hostElement.appendChild(this.cardHost);
+  }
+
+  public initialize() {
+    super.initialize();
+
+    Adaptive.GlobalSettings.useMarkdownInRadioButtonAndCheckbox = false;
+  }
+
+  private parsePadding(source: any): Adaptive.PaddingDefinition {
+    if (source) {
+      if (typeof source === "string") {
+        const uniformPadding = Adaptive.parseEnum(Adaptive.Spacing, source, Adaptive.Spacing.None);
+
+        return new Adaptive.PaddingDefinition(
+          uniformPadding,
+          uniformPadding,
+          uniformPadding,
+          uniformPadding
+        );
+      } else if (typeof source === "object") {
+        return new Adaptive.PaddingDefinition(
+          Adaptive.parseEnum(Adaptive.Spacing, source["top"], Adaptive.Spacing.None),
+          Adaptive.parseEnum(Adaptive.Spacing, source["right"], Adaptive.Spacing.None),
+          Adaptive.parseEnum(Adaptive.Spacing, source["bottom"], Adaptive.Spacing.None),
+          Adaptive.parseEnum(Adaptive.Spacing, source["left"], Adaptive.Spacing.None)
+        );
+      }
+    }
+
+    return null;
+  }
+
+  public parseElement(
+    element: Adaptive.CardElement,
+    source: any,
+    context: Adaptive.SerializationContext
+  ) {
+    if (element instanceof Adaptive.Container && typeof source["rtl"] === "boolean") {
```

Comment:
This isn't necessary. The `rtl` property has been an official AC property for a while and is parsed automatically.

Reply:
Removed.